library/alloc/src/string.rs

```diff
@@ -1553,18 +1553,21 @@ impl String {
         // Replace_range does not have the memory safety issues of a vector Splice.
         // of the vector version. The data is just plain bytes.
 
-        match range.start_bound() {
+        let start = range.start_bound();
```

Comment:
I think comments explaining this change might be beneficial here as well. With or without a big `// WARNING: do not inline this variable` :)